### ISSUE TITLE
Make ADDRESS configurable

### ISSUE THREAD

**Issue by spicy-sauce** [guyskk/pybeautifier#1]
As ADDRESS is hardcoded set to 127.0.0..1:36805, it might be an issue when:
1. Working with containers where 127.0.0.1 is not accessible (should be 0.0.0.0)
2. Port is already taken.

**Comment by guyskk** [comment]
Hello Yossi,

I think the node.js client is in atom-beautify repo, but I'm not very sure because I write it several years ago.

And to make ADDRESS configurable, we can change the code to read host and port value from environ variables, use os.getenv and fall back to hard code value.

Do you able to create PR for it? I'm glad to accept contributions.

Best regards,
guyskk
---Original---
From: "Yossi Shirizli"<yshirizli@gmail.com>
Date: Fri, Jun 4, 2021 03:59 AM
To
Subject: PyBeautifier

Hi there,
I came across this repo, and it seems exactly what I was looking for. 
One thing that I couldn't find there is the node client you've mentioned in the readme: 
So I write the beautifier server in python and *beautifier client in node.js*, they communicate by tcp socket. It's performance is much better(about 300 ms).

What am I missing?

Thanks in advance,
Yossi

**Comment by guyskk** [comment]
Here is the code:
https://github.com/Glavin001/atom-beautify/search?q=pybeautifier&type=commits

And I see some issues related to pybeautifier, but I didn't notice them for many years 😂

**Comment by guyskk** [comment]
And I already changed my editor to vscode for several years, so ......

**Comment by spicy-sauce** [comment]
> Hello Yossi,
> 
> I think the node.js client is in atom-beautify repo, but I'm not very sure because I write it several years ago.

Well, this can be implemented easily, here what I used:
```
    const net = require("net");
    const client = new net.Socket();

    return new Promise(async (resolve, reject) => {
      client.connect(36805, "beautifier", () => {
        console.log("Connected");
        client.write(
          JSON.stringify({
            data: code,
            formaters: [
              {
                name: "autopep8",
                config: {},
              },
            ],
          })
        );
      });

      client.on("data", (data) => {
        try {
          const response = JSON.parse(data.toString());
          if (response.error) reject(response.error);
          resolve(response.data);
        } catch (e) {
          console.error(e);
          reject(e);
        }
        client.destroy(); // kill client after server's response
      });

      client.on("error", (error) => {
        reject(error);
      });

      client.on("close", () => {
        console.debug("Connection closed");
      });
    });
```

> 
> And to make ADDRESS configurable, we can change the code to read host and port value from environ variables, use os.getenv and fall back to hard code value.
> 
> Do you able to create PR for it? I'm glad to accept contributions.
> 

I'll be glad to, I'll provide such a PR soon.

Thanks!

**Comment by spicy-sauce** [comment]
> And I already changed my editor to vscode for several years, so ......

Yes, I use it too, but your repo helps me to provide a formatted python code online (I mean as a service, rather than just running autopep8).

**Comment by guyskk** [comment]
👍👍👍



---Original---
From: "Yossi ***@***.***&gt;
Date: Sun, Jun 6, 2021 23:49 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [guyskk/pybeautifier] Make ADDRESS configurable (#1)




  
Hello Yossi,
 
I think the node.js client is in atom-beautify repo, but I'm not very sure because I write it several years ago.
  
Well, this can be implemented easily, here what I used:
     const net = require("net");     const client = new net.Socket();     return new Promise(async (resolve, reject) =&gt; {       client.connect(36805, "beautifier", () =&gt; {         console.log("Connected");         client.write(           JSON.stringify({             data: code,             formaters: [               {                 name: "autopep8",                 config: {},               },             ],           })         );       });       client.on("data", (data) =&gt; {         try {           const response = JSON.parse(data.toString());           if (response.error) reject(response.error);           resolve(response.data);         } catch (e) {           console.error(e);           reject(e);         }         client.destroy(); // kill client after server's response       });       client.on("error", (error) =&gt; {         reject(error);       });       client.on("close", () =&gt; {         console.debug("Connection closed");       });     });   
And to make ADDRESS configurable, we can change the code to read host and port value from environ variables, use os.getenv and fall back to hard code value.
 
Do you able to create PR for it? I'm glad to accept contributions.
  
I'll be glad to, I'll provide such a PR soon.
 
Thanks!

**Comment by guyskk** [comment]
Fixed via #2, #3 and just released v0.1.2 to PYPI. @spicy-sauce
